profiles/tests/test_profiles_graphql_gdpr_api.py

```diff
@@ -569,3 +585,66 @@ def test_api_tokens_missing(user_gql_client, service_1, query_or_delete, mocker)
         executed = user_gql_client.execute(DELETE_MY_PROFILE_MUTATION)
 
     assert_match_error_code(executed, MISSING_GDPR_API_TOKEN_ERROR)
+
+
+@pytest.mark.parametrize("dry_run", [True, False])
+def test_user_can_delete_data_from_a_service(
+    user_gql_client, service_1, service_2, mocker, dry_run
+):
+    query = DELETE_SERVICE_DATA_MUTATION_TEMPLATE.substitute(
+        service_name=service_1.name, dry_run=str(dry_run).lower()
```

Comment:
Instead of string template manipulation, it's also possible to just pass input variables to GraphQL client and let it handle them. That also makes it easier to test for example input value defaults: don't give some input value at all and see that the used default is what is expected. See e.g. how `CLAIM_PROFILE_MUTATION` is used in `profiles/tests/test_gql_claim_profile_mutation.py` for an example.

And you can also pass Python booleans as is for boolean input variables :slightly_smiling_face:

Reply:
Thanks. I didn't remember that style at all. 😄